pkg/controller/gitopsservice/gitopsservice_controller_test.go

```diff
@@ -0,0 +1,40 @@
+package gitopsservice
+
+import (
+	"os"
+	"testing"
+
+	pipelinesv1alpha1 "github.com/redhat-developer/gitops-operator/pkg/apis/pipelines/v1alpha1"
+)
+
+func TestImageFromEnvVariable(t *testing.T) {
+	cr := &pipelinesv1alpha1.GitopsService{}
+
+	t.Run("Image present as env variable", func(t *testing.T) {
+		image := "quay.io/org/test"
+		os.Setenv(backendImageEnvName, image)
```

Comment:
You should be cleaning this up here (resetting it back to the original value).

Reply:
Updated. Using defer to unset env variable